systemtest/src/test/java/io/strimzi/systemtest/upgrade/AbstractKRaftUpgradeST.java

```diff
@@ -110,66 +110,73 @@ public class AbstractKRaftUpgradeST extends AbstractST {
+        // 2. Send messages
+        produceMessagesAndVerify(testStorage);
 
+        // 3. Make snapshots
         makeComponentsSnapshots(testStorage.getNamespaceName());
         logComponentsPodImagesWithConnect(testStorage.getNamespaceName());
 
-        // Verify FileSink KafkaConnector before upgrade
-        String connectorPodName = kubeClient().listPods(testStorage.getNamespaceName(), Collections.singletonMap(Labels.STRIMZI_KIND_LABEL, KafkaConnect.RESOURCE_KIND)).get(0).getMetadata().getName();
-        KafkaConnectUtils.waitForMessagesInKafkaConnectFileSink(testStorage.getNamespaceName(), connectorPodName, DEFAULT_SINK_FILE_PATH, testStorage.getMessageCount());
+        // 4. Verify KafkaConnector FileSink
+        verifyKafkaConnectorFileSink(testStorage);
 
-        // Upgrade CO to HEAD and wait for readiness of ClusterOperator
+        // 5. Upgrade CO to HEAD and wait for readiness of ClusterOperator
         changeClusterOperator(clusterOperatorNamespaceName, testStorage.getNamespaceName(), upgradeDowngradeData);
 
-        if (TestKafkaVersion.supportedVersionsContainsVersion(upgradeKafkaVersion.getVersion())) {
-            // Verify that Kafka and Connect Pods Rolled
-            waitForKafkaClusterRollingUpdate(testStorage.getNamespaceName());
-            connectPods = RollingUpdateUtils.waitTillComponentHasRolled(testStorage.getNamespaceName(), connectLabelSelector, 1, connectPods);
-            KafkaConnectorUtils.waitForConnectorReady(testStorage.getNamespaceName(), CLUSTER_NAME);
-        }
-
+        // 6. Wait for components to roll
+        maybeWaitForRollingUpdate(testStorage, upgradeKafkaVersion);
         logComponentsPodImagesWithConnect(testStorage.getNamespaceName());
 
-        // Upgrade/Downgrade kafka
-        changeKafkaVersion(testStorage.getNamespaceName(), upgradeDowngradeData);
-        changeKafkaVersionInKafkaConnect(testStorage.getNamespaceName(), upgradeDowngradeData);
-
-        logComponentsPodImagesWithConnect(testStorage.getNamespaceName());
-        checkAllComponentsImages(testStorage.getNamespaceName(), upgradeDowngradeData);
+        verifyPostUpgradeOrDowngradeProcedure(testStorage, upgradeDowngradeData);
+    }
 
+    /**
+     * Verifies the environment after an upgrade or downgrade procedure
+     * by sending new messages, checking connector output, stability, and final state.
+     *
+     * @param testStorage          Test-related configuration and storage
+     * @param upgradeDowngradeData Bundle version modification data
+     */
+    private void verifyPostUpgradeOrDowngradeProcedure(final TestStorage testStorage,
+                                                       final BundleVersionModificationData upgradeDowngradeData) {
+        final KafkaClients clients = ClientUtils.getInstantTlsClientBuilder(testStorage, KafkaResources.tlsBootstrapAddress(CLUSTER_NAME))
+            .withNamespaceName(testStorage.getNamespaceName())
```

Comment:
I know it was there before, but I think that we don't need to specify the Namespace here, as it's anyway specified in `getInstantTlsClientBuilder`